index.js

```diff
@@ -22,24 +22,13 @@ function BinarySplit (matcher) {
 
     while (buf) {
       var idx = firstMatch(buf, offset)
-      if (idx) {
-        var line = bops.subarray(buf, offset, idx)
-        if (idx === buf.length) {
-          buffered = line
-          buf = undefined
-          offset = idx
-        } else {
-          this.push(line)
-          offset = idx + matcher.length
-        }
-      } else if (idx === 0) {
-        buf = bops.subarray(buf, offset + matcher.length)
+      if (typeof idx === 'number' && idx < buf.length) {
```

Comment:
idx can be either `false` or number, so typeof is not necessary here — just `idx !== false`. Although for consistency with JS `indexOf`, I'd return `-1` instead of `false`.